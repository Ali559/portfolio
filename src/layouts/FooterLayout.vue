<script setup lang="ts">
import Button from '@/components/ui/button/Button.vue'
import Separator from '@/components/ui/separator/Separator.vue'
import { allSocials } from '@/data'
import { Code2, Github, Linkedin, Mail } from 'lucide-vue-next'
</script>
<template>
  <footer class="py-8 px-4 sm:px-6 lg:px-8 border-t border-border">
    <div class="max-w-6xl mx-auto">
      <Separator class="mb-8" />
      <div class="flex flex-col sm:flex-row justify-between items-center gap-4">
        <div class="flex items-center space-x-2">
          <Code2 class="h-6 w-6 text-primary" />
          <span>Â© 2025 Ali Barznji. All rights reserved.</span>
        </div>
        <div class="flex space-x-4">
          <Button variant="ghost" size="icon" asChild>
            <a :href="allSocials.github.link" target="_blank" rel="noopener noreferrer">
              <Github class="h-5 w-5" />
            </a>
          </Button>
          <Button variant="ghost" size="icon" asChild>
            <a :href="allSocials.linkedin.link" target="_blank" rel="noopener noreferrer">
              <Linkedin class="h-5 w-5" />
            </a>
          </Button>
          <Button variant="ghost" size="icon" asChild>
            <a :href="`mailto:${allSocials.email.link}`">
              <Mail class="h-5 w-5" />
            </a>
          </Button>
        </div>
      </div>
    </div>
  </footer>
</template>
