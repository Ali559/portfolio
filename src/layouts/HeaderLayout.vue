<script setup lang="ts">
import Button from '@/components/ui/button/Button.vue'
import { navLinks } from '@/data'
import { Code2, Menu, Moon, Sun, X } from 'lucide-vue-next'

defineProps<{
  isDark: boolean
  isMenuOpen: boolean
  activeSection: string
}>()
const emit = defineEmits<{
  (e: 'set-theme', value: boolean): void
  (e: 'toggle-menu', value: boolean): void
  (e: 'scroll-to-section', sectionId: string): void
}>()
</script>
<template>
  <nav
    v-motion
    :initial="{ y: -100 }"
    :enter="{ y: 0, animation: 'ease-in-out', animationDuration: '500ms' }"
    class="fixed top-0 left-0 right-0 z-50 bg-background/80 backdrop-blur-md border-b border-border"
  >
    <div class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
      <div class="flex justify-between items-center h-16">
        <div
          class="flex items-center space-x-2 transition-all duration-300 ease-in-out hover:scale-[105%]"
        >
          <Code2 class="h-8 w-8 text-primary" />
          <span class="text-xl font-medium">Ali Barznji</span>
        </div>

        <!-- Desktop Navigation -->
        <div class="hidden md:flex items-center space-x-8">
          <button
            v-for="(link, index) in navLinks"
            :key="index"
            class="capitalize hover:text-primary transition-colors"
            :class="
              activeSection.toLowerCase() === link.name.toLowerCase()
                ? 'text-primary'
                : 'text-muted-foreground'
            "
            @click="emit('scroll-to-section', link.name.toLowerCase())"
          >
            {{ link.name }}
          </button>
          <Button variant="ghost" size="icon" class="ml-4" @click="emit('set-theme', !isDark)">
            <Sun class="h-5 w-5" v-if="!isDark" />
            <Moon class="h-5 w-5" v-else />
          </Button>
        </div>

        <!-- Mobile Menu Button -->
        <div class="md:hidden flex items-center space-x-2">
          <Button variant="ghost" size="icon" @click="emit('set-theme', !isDark)">
            <Sun class="h-5 w-5" v-if="!isDark" />
            <Moon class="h-5 w-5" v-else />
          </Button>
          <Button variant="ghost" size="icon" @click="emit('toggle-menu', !isMenuOpen)">
            <X class="h-5 w-5" v-if="isMenuOpen" />
            <Menu class="h-5 w-5" v-else />
          </Button>
        </div>
      </div>
      <!-- Mobile Menu -->
      <div
        v-if="isMenuOpen"
        v-motion
        :initial="{ opacity: 0, height: 0 }"
        :exit="{ opacity: 0, height: 0 }"
        :enter="{
          opacity: 1,
          height: 'auto',
        }"
        class="md:hidden py-4 border-t border-border"
      >
        <button
          v-for="(link, index) in navLinks"
          :key="index"
          class="block w-full text-left py-2 capitalize hover:text-primary transition-colors"
          :class="
            activeSection.toLowerCase() === link.name.toLowerCase()
              ? 'text-primary'
              : 'text-muted-foreground'
          "
          @click="emit('scroll-to-section', link.name.toLowerCase())"
        >
          {{ link.name }}
        </button>
      </div>
    </div>
  </nav>
</template>
