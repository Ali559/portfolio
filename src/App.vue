<script setup lang="ts">
import { onBeforeMount, onMounted, ref, watch } from 'vue'
import HeaderLayout from './layouts/HeaderLayout.vue'
import FooterLayout from './layouts/FooterLayout.vue'
import { scrollToSection } from './lib/utils'

const isDark = ref<boolean>(false)

const isMenuOpen = ref<boolean>(false)
const activeSection = ref<string>('home')

onBeforeMount(() => {
  if (localStorage.getItem('isDark') === 'true') {
    isDark.value = true
    document.documentElement.classList.add('dark')
  } else {
    isDark.value = false
    document.documentElement.classList.remove('dark')
  }
})

watch(isDark, (value) => {
  if (value) {
    localStorage.setItem('isDark', 'true')
    document.documentElement.classList.add('dark')
  } else {
    localStorage.setItem('isDark', 'false')
    document.documentElement.classList.remove('dark')
  }
})
</script>

<template>
  <div class="min-h-screen bg-background text-foreground">
    <HeaderLayout
      :active-section="activeSection"
      :is-dark="isDark"
      :is-menu-open="isMenuOpen"
      @set-theme="(value) => (isDark = value)"
      @toggle-menu="isMenuOpen = !isMenuOpen"
      @scroll-to-section="
        (id) => {
          activeSection = id
          scrollToSection(id)
        }
      "
    />

    <main>
      <router-view @change-section="(id: string) => (activeSection = id)" />
    </main>
  </div>

  <FooterLayout />
</template>

<style scoped></style>
