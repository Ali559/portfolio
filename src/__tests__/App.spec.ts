import { describe, it, expect, beforeEach } from 'vitest'

import { flushPromises, mount } from '@vue/test-utils'
import App from '../App.vue'
import HeaderLayout from '@/layouts/HeaderLayout.vue';
import FooterLayout from '@/layouts/FooterLayout.vue';

describe('App', () => {
  let wrapper: ReturnType<typeof mount>;
  beforeEach(async () => {
    wrapper = mount(App, {
      global: {
        stubs: {
          HeaderLayout,
          FooterLayout,
        }
      }
    })
    await flushPromises()
  })
  it('mounts renders properly', () => {
    expect(wrapper.findComponent(HeaderLayout).exists()).toBe(true)
    expect(wrapper.findComponent(FooterLayout).exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('handles dark/light and the mode persits in localStorage mode', async () => {
    const header = wrapper.findComponent(HeaderLayout)
    header.vm.$emit('set-theme', true)
    await flushPromises()

    expect(document.documentElement.classList.contains('dark')).toBe(true)
    expect(localStorage.getItem('isDark')).toBe('true')
    header.vm.$emit('set-theme', false)
    await flushPromises()

    expect(document.documentElement.classList.contains('dark')).toBe(false)
    expect(localStorage.getItem('isDark')).toBe('false')
  })


  it('handles menu open/close in mobile view', async () => {
    const header = wrapper.findComponent(HeaderLayout)
    header.vm.$emit('toggle-menu', true)
    await flushPromises()

    expect(header.vm.isMenuOpen).toBe(true)

    header.vm.$emit('toggle-menu', false)
    await flushPromises()

    expect(header.vm.isMenuOpen).toBe(false)
  })

  it('handles scroll to section', async () => {
    const header = wrapper.findComponent(HeaderLayout)
    header.vm.$emit('scroll-to-section', 'about')
    await flushPromises()

    expect(window.scrollY).toBe(0)
  })

})
