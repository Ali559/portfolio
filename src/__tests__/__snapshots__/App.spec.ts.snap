// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > mounts renders properly 1`] = `
"<div class="min-h-screen bg-background text-foreground">
  <nav initial="[object Object]" enter="[object Object]" class="fixed top-0 left-0 right-0 z-50 bg-background/80 backdrop-blur-md border-b border-border">
    <div class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
      <div class="flex justify-between items-center h-16">
        <div class="flex items-center space-x-2 transition-all duration-300 ease-in-out hover:scale-[105%]"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-code-xml-icon lucide-code-xml h-8 w-8 text-primary">
            <path d="m18 16 4-4-4-4"></path>
            <path d="m6 8-4 4 4 4"></path>
            <path d="m14.5 4-5 16"></path>
          </svg><span class="text-xl font-medium">Ali Barznji</span></div><!-- Desktop Navigation -->
        <div class="hidden md:flex items-center space-x-8"><button class="capitalize hover:text-primary transition-colors text-primary">Home</button><button class="capitalize hover:text-primary transition-colors text-muted-foreground">About</button><button class="capitalize hover:text-primary transition-colors text-muted-foreground">Projects</button><button class="capitalize hover:text-primary transition-colors text-muted-foreground">Experience</button><button class="capitalize hover:text-primary transition-colors text-muted-foreground">Contact</button><button data-slot="button" class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg:not([class*='size-'])]:size-4 shrink-0 [&amp;_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 size-9 ml-4"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-sun-icon lucide-sun h-5 w-5">
              <circle cx="12" cy="12" r="4"></circle>
              <path d="M12 2v2"></path>
              <path d="M12 20v2"></path>
              <path d="m4.93 4.93 1.41 1.41"></path>
              <path d="m17.66 17.66 1.41 1.41"></path>
              <path d="M2 12h2"></path>
              <path d="M20 12h2"></path>
              <path d="m6.34 17.66-1.41 1.41"></path>
              <path d="m19.07 4.93-1.41 1.41"></path>
            </svg></button></div><!-- Mobile Menu Button -->
        <div class="md:hidden flex items-center space-x-2"><button data-slot="button" class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg:not([class*='size-'])]:size-4 shrink-0 [&amp;_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 size-9"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-sun-icon lucide-sun h-5 w-5">
              <circle cx="12" cy="12" r="4"></circle>
              <path d="M12 2v2"></path>
              <path d="M12 20v2"></path>
              <path d="m4.93 4.93 1.41 1.41"></path>
              <path d="m17.66 17.66 1.41 1.41"></path>
              <path d="M2 12h2"></path>
              <path d="M20 12h2"></path>
              <path d="m6.34 17.66-1.41 1.41"></path>
              <path d="m19.07 4.93-1.41 1.41"></path>
            </svg></button><button data-slot="button" class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg:not([class*='size-'])]:size-4 shrink-0 [&amp;_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 size-9"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-menu-icon lucide-menu h-5 w-5">
              <path d="M4 12h16"></path>
              <path d="M4 18h16"></path>
              <path d="M4 6h16"></path>
            </svg></button></div>
      </div><!-- Mobile Menu -->
      <!--v-if-->
    </div>
  </nav>
  <main>
    <router-view></router-view>
  </main>
</div>
<footer class="py-8 px-4 sm:px-6 lg:px-8 border-t border-border">
  <div class="max-w-6xl mx-auto">
    <div data-orientation="horizontal" role="none" data-slot="separator-root" class="bg-border shrink-0 data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px mb-8"></div>
    <div class="flex flex-col sm:flex-row justify-between items-center gap-4">
      <div class="flex items-center space-x-2"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-code-xml-icon lucide-code-xml h-6 w-6 text-primary">
          <path d="m18 16 4-4-4-4"></path>
          <path d="m6 8-4 4 4 4"></path>
          <path d="m14.5 4-5 16"></path>
        </svg><span>Â© 2025 Ali Barznji. All rights reserved.</span></div>
      <div class="flex space-x-4"><a data-slot="button" class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg:not([class*='size-'])]:size-4 shrink-0 [&amp;_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 size-9" href="github.com/ali559" target="_blank" rel="noopener noreferrer"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-github-icon lucide-github h-5 w-5">
            <path d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"></path>
            <path d="M9 18c-4.51 2-5-2-7-2"></path>
          </svg></a><a data-slot="button" class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg:not([class*='size-'])]:size-4 shrink-0 [&amp;_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 size-9" href="linkedin.com/in/ali-barznji-36b664325" target="_blank" rel="noopener noreferrer"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-linkedin-icon lucide-linkedin h-5 w-5">
            <path d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"></path>
            <rect width="4" height="12" x="2" y="9"></rect>
            <circle cx="4" cy="4" r="2"></circle>
          </svg></a><a data-slot="button" class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg:not([class*='size-'])]:size-4 shrink-0 [&amp;_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 size-9" href="mailto:biz@alibarznji.dev"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-mail-icon lucide-mail h-5 w-5">
            <path d="m22 7-8.991 5.727a2 2 0 0 1-2.009 0L2 7"></path>
            <rect x="2" y="4" width="20" height="16" rx="2"></rect>
          </svg></a></div>
    </div>
  </div>
</footer>"
`;
